src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -168,6 +178,10 @@ public static IResourceBuilder<ProjectResource> AddProject(this IDistributedAppl
     /// </example>
     public static IResourceBuilder<ProjectResource> AddProject(this IDistributedApplicationBuilder builder, [ResourceName] string name, string projectPath, string? launchProfileName)
     {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(name);
```

Comment:
Several of these should probably throw if empty, as in:

```suggestion
        ArgumentException.ThrowIfNullOrEmpty(name);
```

Reply:
They could. However, there is more validations of names that ensures they're not empty, Technically this null check isn't nesseccary, but I like the consistency of checking all references for null in public API calls.